src/librustc_mir/transform/const_prop.rs

```diff
@@ -518,27 +518,28 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 }
             }
 
-            // Work around: avoid ICE in miri.
-            // FIXME(wesleywiser) we don't currently handle the case where we try to make a ref
-            // from a function argument that hasn't been assigned to in this function. The main
-            // issue is if an arg is a fat-pointer, miri `expects()` to be able to read the value
-            // of that pointer to get size info. However, since this is `ConstProp`, that argument
-            // doesn't actually have a backing value and so this causes an ICE.
+            // Work around: avoid ICE in miri. FIXME(wesleywiser)
+            // The Miri engine ICEs when taking a reference to an uninitialized unsized
+            // local. There's nothing it can do here: taking a reference needs an allocation
+            // which needs to know the size. Normally that's okay as during execution
+            // (e.g. for CTFE) it can never happen. But here in const_prop
+            // we leave function arguments uninitialized, so if one of these is unsized
+            // and has a reference taken, we get an ICE.
             Rvalue::Ref(_, _, Place { base: PlaceBase::Local(local), projection: box [] }) => {
                 trace!("checking Ref({:?})", place);
                 let alive =
                     if let LocalValue::Live(_) = self.ecx.frame().locals[*local].value {
                         true
                     } else { false };
 
-                if local.as_usize() <= self.ecx.frame().body.arg_count && !alive {
```

Comment:
This is super subtle. The first argument ID is 1, because the return place's ID is already used. So the original condition was right but additionally was applied to the return place local.

Reply:
Yeah I realized the same thing at the same time. :) Fixed in https://github.com/rust-lang/rust/commit/05e4e6ba0d5970e4150854cd5e6de763c232450b.